core/include/initcall.h

```diff
@@ -30,12 +30,20 @@ struct initcall {
 		{ .func = (fn), }
 #endif
 
+#define preinitcall_begin \
+			SCATTERED_ARRAY_BEGIN(preinitcall, struct initcall)
+#define preinitcall_end SCATTERED_ARRAY_END(preinitcall, struct initcall)
+
 #define initcall_begin	SCATTERED_ARRAY_BEGIN(initcall, struct initcall)
 #define initcall_end	SCATTERED_ARRAY_END(initcall, struct initcall)
 
 #define finalcall_begin	SCATTERED_ARRAY_BEGIN(finalcall, struct initcall)
 #define finalcall_end	SCATTERED_ARRAY_END(finalcall, struct initcall)
 
+#define preinit_early(fn)		__define_initcall(preinit, 1, fn)
```

Comment:
I think you need to document somewhere that this calls are meant to be used by TEE part, not by Nexus

Reply:
Good point, I'll add something.